src/ring_theory/finiteness.lean

```diff
@@ -605,10 +612,58 @@ begin
   simp only [set.image_Union]
 end
 
+end semiring
+
+section ring
+
+variables [comm_ring R] [add_comm_monoid M]
+
+/-- If `add_monoid_algebra R M` is of finite type, there there is a `G : finset M` such that its
+image generates, as algera, `add_monoid_algebra R M`. -/
+lemma generators_of_finite_type : finite_type R (add_monoid_algebra R M) → ∃ G : finset M,
+  algebra.adjoin R (of' R M '' G) = ⊤ :=
```

Comment:
I completely understand the name, but I'm not sure we should deviate from the convention here. How about
```suggestion
lemma exists_finset_adjoin_eq_top : finite_type R (add_monoid_algebra R M) →
  ∃ G : finset M, algebra.adjoin R (of' R M '' G) = ⊤ :=
```
Do you want the finite type assumption to come be TC? Then you should put it in `[]` before the `:` and use `unfreezingI` to destruct it.